File: code/js/src/domain/Team.ts
export interface TeamDomain {
    id: number,
    name: string,
    isCreated: Boolean,
    isClosed: Boolean,
    assignment: number,
}

export class TeamDomain {
    constructor(
        id: number,
        name: string,
        isCreated: Boolean,
        isClosed: Boolean,
        assignment: number,
    ) {
        this.id = id
        this.name = name
        this.isCreated = isCreated
        this.isClosed = isClosed
        this.assignment = assignment
    }
}

File: code/android/app/src/main/java/isel/ps/classcode/domain/Team.kt
package isel.ps.classcode.domain

import isel.ps.classcode.domain.deserialization.ClassCodeTeamDeserialization
import isel.ps.classcode.domain.dto.LocalTeamDto

/**
 * Represents a team
 */
data class Team(
    val id: Int,
    val name: String,
    val isCreated: Boolean,
    val isClosed: Boolean,
    val assignment: Int,
) {
    constructor(classCodeTeamDeserialization: ClassCodeTeamDeserialization) : this(
        id = classCodeTeamDeserialization.id,
        name = classCodeTeamDeserialization.name,
        isCreated = classCodeTeamDeserialization.isCreated,
        isClosed = classCodeTeamDeserialization.isClosed,
        assignment = classCodeTeamDeserialization.assignment,
    )
    val teamSlug = name.replace(" ", "-").lowercase()
    fun toLocalTeamDto(courseId: Int, courseName: String, classroomId: Int): LocalTeamDto = LocalTeamDto(id = id, name = name, isCreated = isCreated, assignment = assignment, courseId = courseId, courseName = courseName, classroomId = classroomId, isClosed = isClosed)
}
